CORTX-33018: Support external ssl secret (#362)
Seagate/cortx-k8s: README.md

| Name                                                  | Description                                                     | Default Value       |
| ----------------------------------------------------- | --------------------------------------------------------------- | ------------------- |
| `common.storage_provisioner_path`                     | TODO       | `/mnt/fs-local-volume` |

> | `common.ssl.external_secret`                          | If specified, a Secret that containes the certificate used for CORTX servers. | `""` |

Reviewer: ```suggestion
| `common.ssl.external_secret`                          | If specified, a Secret that contains the certificate used for CORTX servers. | `""` |
```

| ---------------------------------------------- | ------------------------------------------------------------- |
| `CORTX_DEPLOY_CUSTOM_BLOCK_STORAGE_CLASS`      | When set to a non-empty string, CORTX will skip the deployment of the `cortx-block-data` Helm Chart, along with the automatic creation of the underlying PersistentVolumes necessary to deploy CORTX. It will instead use the value of this variable as the expected Kubernetes StorageClass and expect all available PersistentVolumes to be created manually by the user. See [Advanced Deployment Scenarios - Using manually-created PersistentVolumes](doc/advanced-deployment-scenarios.md#using-manually-created-persistentvolumes) for use case details.  |

### Deploy with SSL Certificate

> By default, CORTX is installed with a self-signed SSL certificate.  This section describes how to deploy with a custom SSL instead.

Author: Aaarg! Thanks for catching this...

> By default, CORTX is installed with a default self-signed SSL certificate.  This section describes how to deploy with a specific SSL certificate.

Reviewer 2: You don't need 'default' twice so close together here. I would change 'specific' in the second sentence to 'custom'.

> By default, CORTX is installed with a specific self-signed SSL certificate.  This section describes how to deploy with a specific SSL certificate.

Reviewer 2: `By default, CORTX is installed with a self-signed SSL certificate.  This section describes how to deploy with a custom SSL certificate instead.`

> By default, CORTX is installed with a specific self-signed SSL certificate.  This section describes how to deploy with a specific SSL certificate.

Reviewer 2: ```suggestion
By default, CORTX is installed with a self-signed SSL certificate.  This section describes how to deploy with a custom SSL certificate instead.
```

> Before deploying CORTX, create a Kubernetes secret that contains the SSL certificate.

Reviewer 2: ```suggestion
Before deploying CORTX, create a Kubernetes Secret that contains the SSL certificate.
```

  * The Secret must contain the key `cortx.pem`
  * The value must be in PEM format and must contain the Private Key and Certificate.

   ```bash
   # where "mycert.pem" contains the certificate

>    kubectl create secret generic my-ssl-cert --from-file=cortx.pem=mycert.pem -n "${namespace}"

Reviewer: ```suggestion
   kubectl create secret generic my-ssl-cert --from-file=cortx.pem=mycert.pem --namespace=${NAMESPACE}"
```

Nitpick, but previous examples in this README use uppercase and the long flag name.

   ```

Specify common.ssl.external_secret in `solution.yaml`:

>    ```bash

Reviewer: ```suggestion
   ```yaml
```

   solution:
     common:
       ssl:
         external_secret: my-ssl-cert
   ```

> When you deploy CORTX it will use the certificate from the Secret `my-ssl-cert`.

Reviewer 2: ```suggestion
When you deploy CORTX, it will now use the SSL certificate from the Secret `my-ssl-cert` instead of the default self-signed SSL certificate.
```

> When you deploy CORTX, it will now use the SSL certificate from the Secret `my-ssl-cert` instead of the efault self-signed SSL certificate.

Reviewer 2: ```suggestion
When you deploy CORTX, it will now use the SSL certificate from the Secret `my-ssl-cert` instead of the default self-signed SSL certificate.
```
